utils/demangle.c

```diff
@@ -1464,6 +1465,15 @@ static int dd_source_name(struct demangle_data *dd)
 	if (add_name)
 		dd_append_len(dd, p, num);
 	__dd_consume_n(dd, num, NULL);
+	while (true) {
+		separator = strstr(dd->new, "..");
+		if (separator) {
+			separator[0] = ':';
+			separator[1] = ':';
+		}
+		else
+			break;
```

Comment:
In general, I prefer reducing the nesting level by checking the break condition early.  Something liike below..

```c
while (true) {
    separator = strstr(dd->new, "..");
    if (separator == NULL)
        break;
    separator[0] = ':';
    separator[1] = ':';
}
```